booking/src/main/java/org/dcsa/conformance/standards/booking/checks/BookingChecks.java

```diff
@@ -0,0 +1,651 @@
+    "validate shipment cutOff Date time code",
+    (body) -> {
+      var shipmentCutOffTimes = body.path("shipmentCutOffTimes");
+      var receiptTypeAtOrigin = body.path("receiptTypeAtOrigin").asText("");
+      var issues = new LinkedHashSet<String>();
+      var cutOffDateTimeCodes = StreamSupport.stream(shipmentCutOffTimes.spliterator(), false)
+        .map(p -> p.path("cutOffDateTimeCode"))
+        .filter(JsonNode::isTextual)
+        .map(n -> n.asText(""))
+        .collect(Collectors.toSet());
+      if (receiptTypeAtOrigin.equals("CFS")) {
+        if (cutOffDateTimeCodes.contains("LCL")) {
+          issues.add("For ReceiptTypeAtOrigin as CFS, the shipment cut off date time must have only LCL");
+        }
+      }
+      return issues;
+    }
+  );
+
+  private static final Consumer<MultiAttributeValidator> ALL_AMF = (mav) -> mav.submitAllMatching("advanceManifestFilings.*");
+
+  private static final JsonContentCheck ADVANCED_MANIFEST_FILING_CODES_UNIQUE = JsonAttribute.allIndividualMatchesMustBeValid(
+    "The combination of 'countryCode' and 'manifestTypeCode' in 'advanceManifestFilings' must be unique",
+    ALL_AMF,
+    JsonAttribute.unique("countryCode", "manifestTypeCode")
+  );
+  private static final Consumer<MultiAttributeValidator> ALL_SHIPMENT_CUTOFF_TIMES = (mav) -> mav.submitAllMatching("shipmentCutOffTimes.*");
+
+  private static final JsonContentCheck SHIPMENT_CUTOFF_TIMES_UNIQUE = JsonAttribute.allIndividualMatchesMustBeValid(
+    "in 'shipmentCutOfftimes' cutOff date time code must be unique",
+    ALL_SHIPMENT_CUTOFF_TIMES,
+    JsonAttribute.unique("cutOffDateTimeCode")
+  );
+  private static final JsonContentCheck AMF_CC_MTC_COMBINATION_VALIDATIONS = JsonAttribute.allIndividualMatchesMustBeValid(
+    "Validate combination of 'countryCode' and 'manifestTypeCode' in 'advanceManifestFilings'",
+    ALL_AMF,
+    JsonAttribute.combineAndValidateAgainstDataset(BookingDataSets.AMF_CC_MTC_COMBINATIONS, "countryCode", "manifestTypeCode")
+  );
+
+  private static final JsonContentCheck VALIDATE_SHIPMENT_LOCATIONS = JsonAttribute.customValidator(
+    "Validate shipmentLocations",
+    body -> {
+      var issues = new LinkedHashSet<String>();
+      var shipmentLocations = body.path("shipmentLocations");
+      var receiptTypeAtOrigin = body.path("receiptTypeAtOrigin").asText("");
+      var polNode =
+        StreamSupport.stream(shipmentLocations.spliterator(), false)
+          .filter(o ->  o.path("locationTypeCode").asText("").equals("POL")
+            || o.path("locationTypeCode").asText("").equals("PDE") )
+          .findFirst()
+          .orElse(null);
+      var podNode =
+        StreamSupport.stream(shipmentLocations.spliterator(), false)
+          .filter(o -> o.path("locationTypeCode").asText("").equals("POD")
+          || o.path("locationTypeCode").asText("").equals("PRE") )
+          .findFirst()
+          .orElse(null);
```

Comment:
I do not understand why `POL` and `PDE` are paired and ditto for `PRE` and `POD`.  Normally, the pairings are `PRE` + `POL` vs. `POD` + `PDE`.